### PR TITLE
fix metadata generator

### PR DESCRIPTION
**Pull request by strofimo**

### CODE REVIEW

File: src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-    {MemberType: 16,Name: 'Other', Type: module.MyClass_$1},
-    {MemberType: 16,Name: 'Getter', Type: Number},
-    {MemberType: 8,Name: 'method', Type: Void},
-    {MemberType: 8,Name: 'method2', Type: Object},
-    {MemberType: 8,Name: 'method3', Type: ss.IEnumerable},
-    {MemberType: 8,Name: 'method4', Type: ss.IList_$1},
-    {MemberType: 8,Name: '_assert', Type: Void},
-    {MemberType: 8,Name: 'get_item', Type: Object},
-    {MemberType: 8,Name: 'set_item', Type: Object}];
+    {MemberType: 16,Name: 'Other', Type: eval("try{module.MyClass_$1;}catch{}"},
```
Comment (isc30) [inline src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt:7]: missing `)` in the end

Comment (isc30) [reply src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt:7]: semicolon can be removed it can be `eval("try{module.MyClass_$1}catch{})"`

**Review by isc30 — CHANGES_REQUESTED** [review]

File: src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt (hunk 1 above)
Comment (strofimo) [reply src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt]: fixed

**Review by isc30 — APPROVED** [review]

File: src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt [hunk 2]
```diff
@@ -1,19 +1,20 @@
 ss.registerMetadataImporter(function(){(function(ss) {
   "use strict"
   var Void = null;
+  var Type = function(){};
```
Comment (KevReed) [inline src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt:4]: this should just map to Function

**Review by KevReed — CHANGES_REQUESTED** [review]

File: src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt (hunk 2 above)
Comment (strofimo) [reply src/DSharp.Compiler.Tests/Source/Type/Reflection/Metadata.txt]: Fixed

**Review by KevReed — APPROVED** [review]

**Pull request merged**